Replicator/tests/ReplParams.cc

```diff
@@ -0,0 +1,129 @@
+//
+// Created by Callum Birks on 07/11/2022.
+//
+
+#include "ReplParams.hh"
+
+ReplParams::ReplParams(std::vector<C4ReplicationCollection> collections_, const AllocedDict& options,
+                       StatusCallback statusCallback, DocsEndedCallback docsEndedCallback,
+                       BlobProgressCallback blobProgressCallback, EncryptionCallback encryptionCallback,
+                       DecryptionCallback decryptionCallback, void *callbackContext_, C4SocketFactory *socketFactory_)
+{
+    _collectionVector = std::vector<C4ReplicationCollection>(collections_);
+    collections = _collectionVector.data();
+    collectionCount = _collectionVector.size();
+    _optionsDict = options;
+    optionsDictFleece = _optionsDict.data();
+    onStatusChanged = statusCallback;
+    onDocumentsEnded = docsEndedCallback;
+    propertyEncryptor = encryptionCallback;
+    propertyDecryptor = decryptionCallback;
+    callbackContext = callbackContext_;
+    socketFactory = socketFactory_;
+}
+
+ReplParams::ReplParams(const ReplParams &other) {
+    _collectionVector = { other._collectionVector };
+    _collectionsOptionsDict = { other._collectionsOptionsDict };
+    _paramSetterOptions = { other._paramSetterOptions };
+    _optionsDict = AllocedDict(other._optionsDict);
+    collections = _collectionVector.data();
+    collectionCount = other.collectionCount;
+    optionsDictFleece = _optionsDict.data();
+    onStatusChanged = other.onStatusChanged;
+    onDocumentsEnded = other.onDocumentsEnded;
+    onBlobProgress = other.onBlobProgress;
+    propertyEncryptor = other.propertyEncryptor;
+    propertyDecryptor = other.propertyDecryptor;
+    callbackContext = other.callbackContext;
+    socketFactory = other.socketFactory;
+}
+
+void ReplParams::addCollections(std::vector<C4ReplicationCollection> collectionsToAdd) {
+    for(const auto& c : collectionsToAdd) {
+        _collectionVector.push_back(c);
+    }
+    collections = _collectionVector.data();
+    collectionCount = _collectionVector.size();
+}
+
+AllocedDict ReplParams::setOptions(const AllocedDict& params, const AllocedDict& options) {
+    AllocedDict result = params;
+    for(Dict::iterator i(options); i; ++i) {
+        result = repl::Options::updateProperties(result, i.keyString(), i.value());
+    }
+    return result;
```

Comment:
Who is going to hold it? Should this class do?
I see, it is to be used internally for other setOptions methods. More apt as a private static method.

Reply:
Yeah it is currently used only internally, I left it public simply to give a user a simple method for updating a dict. Although you're probably right, should be a private method.